AwsEncryptionSDK/dafny/AwsEncryptionSdk/src/Index.dfy

```diff
@@ -23,15 +23,20 @@ module
   }
 
   method ESDK(config: AwsEncryptionSdkConfig)
-    returns (res: Result<ESDKClient, Error>)
+    returns (res: Result<IAwsEncryptionSdkClient, Error>)
```

Comment:
Why? The caller can go down, but the class will have more detailed options.
Is `smithy-dafny` did it this way, then you don't need to do the work on 29-32.
But if we MUST, then at least we need an `ensures` clause to that callers _could_ do the same.

Reply:
The `smithy-dafny` change causing this was here: https://github.com/smithy-lang/smithy-dafny/pull/301/files#r1406736341

I think we collectively recognized that it was a Dafny-only breaking change but a good change to make. It's been a real pain to adjust for though and I regret not trying harder to avoid it.

Reply:
sigh